test/wanda/executions/evaluation_test.exs

```diff
@@ -1188,4 +1188,93 @@ defmodule Wanda.Executions.EvaluationTest do
              } = Evaluation.execute(execution_id, group_id, checks, gathered_facts, env)
     end
   end
+
+  describe "failure message is evaluated" do
+    test "should return an evaluated failure message inside the expectation evaluation when the result is false, otherwise a nil field" do
+      execution_id = UUID.uuid4()
+      group_id = UUID.uuid4()
+      fact_value = Enum.random(1..10)
+      incorrect_fact_value = Enum.random(11..20)
+
+      [%Catalog.Fact{name: fact_name}] = catalog_facts = build_list(1, :catalog_fact)
```

Comment:
I noticed that you use `build_list(1, ...)` a few times here; Wouldn't it be clearer to just use `build` and wrap around `[]` when you need that to be an array?

e.g.:
```
%Catalog.Fact{name: fact_name} = catalog_fact = build(1, :catalog_fact)
....
      [%Catalog.Check{id: check_id}] =
        checks =
        build_list(1, :check,
          severity: :critical,
          facts: [catalog_fact],
          values: [],
          expectations: expectations
        )
 ```
 Not a strong opinion though, and I might be missing something :smile: